massa-bootstrap/src/server_binder.rs

```diff
@@ -1,57 +1,54 @@
 // Copyright (c) 2022 MASSA LABS <info@massa.net>
 
 use crate::error::BootstrapError;
-use crate::establisher::Duplex;
 use crate::messages::{
     BootstrapClientMessage, BootstrapClientMessageDeserializer, BootstrapServerMessage,
     BootstrapServerMessageSerializer,
 };
 use crate::settings::BootstrapSrvBindCfg;
-use async_speed_limit::clock::StandardClock;
-use async_speed_limit::{Limiter, Resource};
 use massa_hash::Hash;
 use massa_hash::HASH_SIZE_BYTES;
 use massa_models::serialization::{DeserializeMinBEInt, SerializeMinBEInt};
 use massa_models::version::{Version, VersionDeserializer, VersionSerializer};
 use massa_serialization::{DeserializeError, Deserializer, Serializer};
 use massa_signature::KeyPair;
 use massa_time::MassaTime;
-use std::convert::TryInto;
-use std::net::SocketAddr;
-use std::thread;
-use std::time::Duration;
-use tokio::io::{AsyncReadExt, AsyncWriteExt};
-use tokio::runtime::Handle;
-use tokio::time::error::Elapsed;
+use std::{
+    convert::TryInto,
+    io::{ErrorKind, Read, Write},
+    net::{SocketAddr, TcpStream},
+    thread,
+    time::Duration,
+};
 use tracing::error;
 
 /// Bootstrap server binder
-pub struct BootstrapServerBinder<D: Duplex> {
+pub struct BootstrapServerBinder {
     max_bootstrap_message_size: u32,
     max_consensus_block_ids: u64,
     thread_count: u8,
     max_datastore_key_length: u8,
     randomness_size_bytes: usize,
     size_field_len: usize,
     local_keypair: KeyPair,
-    duplex: Resource<D, StandardClock>,
+    duplex: TcpStream,
     prev_message: Option<Hash>,
     version_serializer: VersionSerializer,
     version_deserializer: VersionDeserializer,
     write_error_timeout: MassaTime,
 }
 
-impl<D: Duplex + 'static> BootstrapServerBinder<D> {
+impl BootstrapServerBinder {
     /// Creates a new `WriteBinder`.
     ///
     /// # Argument
     /// * `duplex`: duplex stream.
     /// * `local_keypair`: local node user keypair
     /// * `limit`: limit max bytes per second (up and down)
     #[allow(clippy::too_many_arguments)]
-    pub fn new(duplex: D, local_keypair: KeyPair, cfg: BootstrapSrvBindCfg) -> Self {
+    pub fn new(duplex: TcpStream, local_keypair: KeyPair, cfg: BootstrapSrvBindCfg) -> Self {
         let BootstrapSrvBindCfg {
-            max_bytes_read_write: limit,
+            max_bytes_read_write: _limit,
```

Comment:
unused _limit ?

Reply:
Ok, thank you for your explanation. If we keep it like this, maybe we can add a comment to explain with it is not used for now

Reply:
Yes limiter isn' a priority thing but should be done for sure !